fem/gridfunc.hpp

```diff
@@ -500,49 +500,120 @@ class GridFunction : public Vector
                                  const Array<int> *elems = NULL) const;
 
-   /// Returns ||grad u_ex - grad u_h||_L2 in element ielem for H1 or L2 elements
+   /// @brief Returns ||grad u_ex - grad u_h||_L2 in element ielem for H1 or L2 elements
+   ///
+   /// @note Quadratures with negative weights (as in some simplex integration
+   ///       rules in MFEM) can produce negative integrals even with
+   ///       non-negative integrands. To avoid returning negative errors this
+   ///       function uses the absolute values of the element-wise integrals.
+   ///       This may lead to results which are not entirely consistent with
+   ///       such integration rules.
    virtual real_t ComputeElementGradError(int ielem, VectorCoefficient *exgrad,
                                           const IntegrationRule *irs[] = NULL) const;
 
-   /// Returns ||u_ex - u_h||_L2 for H1 or L2 elements
-   /* The @a elems input variable expects a list of markers:
-      an elem marker equal to 1 will compute the L2 error on that element
-      an elem marker equal to 0 will not compute the L2 error on that element */
+   /// @brief Returns ||u_ex - u_h||_L2 for H1 or L2 elements
+   ///
+   /// The @a elems input variable expects a list of markers:
+   /// an elem marker equal to 1 will compute the L2 error on that element
+   /// an elem marker equal to 0 will not compute the L2 error on that element
+   ///
+   /// @note Quadratures with negative weights (as in some simplex integration
+   ///       rules in MFEM) can produce negative integrals even with
+   ///       non-negative integrands. To avoid returning negative errors this
+   ///       function uses the absolute values of the element-wise integrals.
+   ///       This may lead to results which are not entirely consistent with
+   ///       such integration rules.
    virtual real_t ComputeL2Error(Coefficient &exsol,
                                  const IntegrationRule *irs[] = NULL,
                                  const Array<int> *elems = NULL) const
    { return GridFunction::ComputeLpError(2.0, exsol, NULL, irs, elems); }
 
+   /// @brief Returns ||u_ex - u_h||_L2 for ND or RT elements
+   ///
+   /// @note Quadratures with negative weights (as in some simplex integration
+   ///       rules in MFEM) can produce negative integrals even with
+   ///       non-negative integrands. To avoid returning negative errors this
+   ///       function uses the absolute values of the element-wise integrals.
+   ///       This may lead to results which are not entirely consistent with
+   ///       such integration rules.
    virtual real_t ComputeL2Error(VectorCoefficient &exsol,
                                  const IntegrationRule *irs[] = NULL,
                                  const Array<int> *elems = NULL) const;
 
-   /// Returns ||grad u_ex - grad u_h||_L2 for H1 or L2 elements
+   /// @brief Returns ||grad u_ex - grad u_h||_L2 for H1 or L2 elements
+   ///
```

Comment:
It would be nice to mention here that the gradient is only over the interior of DGs and one should call `ComputeDGFaceJumpError()` to get the total error of the gradient in the sense of generalized functions or just intuitively.